src/primitives/gf32_ext.rs

```diff
@@ -0,0 +1,394 @@
+        let mut acc = Self::ZERO;
+        for xi in other.inner.iter().rev() {
+            acc.mul_by_ext_elem();
+            acc += self * xi;
+        }
+        acc
+    }
+}
+
+/// The field of order 1024.
+pub type Fe1024 = Fe32Ext<2>;
+
+impl Field for Fe1024 {
+    /// The zero element of the field.
+    const ZERO: Self = Self::new([Fe32::Q, Fe32::Q]);
+
+    /// The one element of the field.
+    const ONE: Self = Self::new([Fe32::P, Fe32::Q]);
+
+    // Chosen somewhat arbitrarily.
+    /// A generator of the field.
+    const GENERATOR: Self = Self::new([Fe32::P, Fe32::H]);
+
+    /// The order of the multiplicative group of the field.
+    ///
+    /// This constant also serves as a compile-time check that we can count
+    /// the entire field using a `usize` as a counter.
+    const MULTIPLICATIVE_ORDER: usize = 1023;
+
+    const MULTIPLICATIVE_ORDER_FACTORS: &'static [i64] = &[1, 3, 11, 31, 33, 93, 341, 1023];
+
+    #[inline]
+    fn _add(&self, other: &Self) -> Self {
+        Self::new([self.inner[0] + other.inner[0], self.inner[1] + other.inner[1]])
+    }
+
+    #[inline]
+    fn _sub(&self, other: &Self) -> Self { self._add(other) }
+
+    #[inline]
+    fn _mul(&self, other: &Self) -> Self { self.mul_by_elem(other) }
+
+    #[inline]
+    fn _div(&self, other: &Self) -> Self { other.multiplicative_inverse() * self }
+
+    #[inline]
+    fn _neg(self) -> Self { self }
+
+    fn multiplicative_inverse(self) -> Self {
+        // Aliases to make the below equations easier to read
+        let a0 = self.inner[0];
+        let a1 = self.inner[1];
+        let p0 = Self::POLYNOMIAL.inner[0];
+        let p1 = Self::POLYNOMIAL.inner[1];
+
+        // Inverse of the 2x2 multiplication matrix defined by a0, a1.
+        let det = (a0 * a0) + (p1 * a0 * a1) + (p0 * a1 * a1);
+        Self::new([(a0 + p1 * a1) / det, (Fe32::Q - a1) / det])
+    }
+}
+super::impl_ops_for_fe!(impl for Fe1024);
+
+impl ExtensionField for Fe1024 {
+    type BaseField = Fe32;
+
+    const DEGREE: usize = 2;
+
+    // Ultimately it doesn't really matter what choice of polynomial we make
+    // here. We choose the value from BIP 93, which we note differs from the
+    // value used in bech32 error correcting code, such as
+    // https://github.com/sipa/bech32/commit/e97932d4c86e343ace49ae6170ae0c4871820152
+    //
+    // (Specifically, the third element of that exp table, 311, expanded into binary,
+    // 01001 10111, and mapped back to Fe32, gives us FH rather than PP. But really,
+    // it doesn't matter, except that it is part of the `Fe1024` API and we cannot
+    // change it once we have published it, since changing it amounts to moving to
+    // a different, though isomorphic, field.)
+    const POLYNOMIAL: Self = Self::new([Fe32::P, Fe32::P]);
+
+    /// The element zeta such that the extension field is defined as GF32\[zeta\].
+    ///
+    /// Alternately, the image of x in the mapping GF32\[x\]/p(x) -> the field
```

Comment:
nit: The escapes make it pretty noisy in the source code, perhaps we just put backtics around the whole expression?

```rust
    /// Alternately, the image of x in the mapping `GF32[x]/p(x)` -> the field
```

(I built the docs and looked at it, looks ok in HTML.)